### ISSUE TITLE
Put bundled steamworks sdk behind feature

### ISSUE THREAD

**Issue by tbillington** [HouraiTeahouse/bevy_steamworks#34]
The bundling of the steamworks dynamic libraries with releases is very convenient, however it should be possible to opt-out if the user so chooses.

My suggestion would be a `bundled` feature that is on by default, but could be disabled through the use of "no default features" in Cargo.toml.

### Reasons users may opt-out

**Supply chain best practice**

There's no "easy" way to verify the library files are the originals from Valve. Made especially difficult because they are not included in the repository, understandably.

Users should be able to provide the libraries themselves for full certainty they're legitimate.

**Patch updates without relying on `bevy_steamworks`**

Users can update the steamworks dynamic libs if they're semver compliant on their own without requiring an upstream patch by `bevy_steamworks`.

---

Prior art in bundling binary libraries: [Rusqlite](https://github.com/rusqlite/rusqlite?tab=readme-ov-file#optional-features) (sqlite).

**Comment by tbillington** [comment]
Another option, though I don't know how preferable this is, is either a build.rs step or shell script in the repo that would fetch from the official steamworks servers eg (https://partner.steamgames.com/downloads/steamworks_sdk_159.zip) and unpack the libraries locally.

That could be a separate feature (`"download_sdk"`?) that users could use in addition to disabling `"bundled"` to automate the process of acquiring the steamworks sdk.

**Comment by james7132** [comment]
This is probably something worth filing an issue with `steamworks-rs`. This crate/repo is merely an integration between it and Bevy.

**Comment by tbillington** [comment]
Makes sense, I've opened an issue upstream https://github.com/Noxime/steamworks-rs/issues/160.

**Comment by james7132** [comment]
If or when the feature is upstream, we'll mirror it in this integration crate. Closing this out for now.
